search-commons/src/main/java/no/unit/nva/search2/common/Query.java

```diff
@@ -108,14 +111,15 @@ protected String toCsvText(SwsResponse response) {
     public PagedSearch toPagedResponse(SwsResponse response) {
         final var requestParameter = toNvaSearchApiRequestParameter();
         final var source = URI.create(getNvaSearchApiUri().toString().split(PATTERN_IS_URL_PARAM_INDICATOR)[0]);
+        final var aggregationFormatted = AggregationFormat.apply(response.aggregations(),aggregationsDef()).toString();
```

Comment:
Isn't AggregationFormat actually "our" format of aggregation returned after we have converted OpenSearch original aggregation to "our own" format? By adding JsonNode aggregation field to AggregationFormat and renaming AggregattionFormat to just Aggregation, we can achieve more sense:

For example:

`Aggregation.fromOpenSearchAggregation(jsonNode).toJsonString()`

I think it makes more sense now since you have extracted `AggregationFormat` from `SwsResponse.class`

Since we know the context we can change it to ResourceAggregation, and tickets will have TicketsAggregation. In that we you do not need to pass aggregationDefs to Aggregation, since Resource Aggregation should know its own definitions.

Reply:
That should be a separate task I think, but yes.